packages/components/tests/unit/helpers/validateCssAndMatchSnapshot.ts

```diff
@@ -1,17 +1,108 @@
 import { expect } from '@jest/globals';
+import { componentMeta } from '@porsche-design-system/component-meta';
+import { getCssObject } from '../../../src/test-utils';
+import { TagName } from '@porsche-design-system/shared';
 
-/*
-  TODO: Existing tests for:
-    - !important on host display style
-    - !important on slotted styles
-    - @hover media query for hover styles
-    - Host hidden styles
-    should also be covered for all getComponentCss conditions
-*/
 export const validateCssAndMatchSnapshot = (css: string) => {
-  // We shouldn't use visibility: visible since it cannot be overwritten, use inherit instead
-  expect(css).not.toMatch(/visibility:\s*(?:var\(.+,\s*visible\s*\)|visible);/);
-  // Invalid css which was produced before
-  expect(css).not.toMatch('. {');
+  const cssObject = getCssObject(css);
+  const componentName = expect.getState().testPath.match(/\/([^\/]+)\/[^\/]+\.spec\.ts/)[1];
+  const componentTagName = `p-${componentName}` as TagName;
+  // Extract componentMeta from testPath, if it's a functional component this will be undefined
+  const currentComponentMeta = componentMeta[componentTagName];
+
+  expect(css).not.toMatch('. {'); // Invalid css which was produced before
+
+  validateVisibilityStyle(cssObject);
+  validateSlottedStyles(cssObject, componentTagName);
+  validateHoverMediaQuery(cssObject);
+
+  // Validations for components only
+  if (currentComponentMeta && !currentComponentMeta.isInternal) {
+    expect(cssObject[':host([hidden])']).toEqual({ display: 'none !important' });
+    validateHostDisplayStyle(cssObject);
+  }
+
   expect(css).toMatchSnapshot();
 };
+
+// We shouldn't use visibility: visible since it cannot be overridden, use inherit instead
+const validateVisibilityStyle = (cssObject: object) => {
+  recursivelyApply(cssObject, (key, value) => {
+    if (key.includes('visibility')) {
+      expect(value).not.toMatch(/visible/);
+    }
+  });
+};
+
+// Expect no !important rule on display style of :host selector since it should be overridable
+const validateHostDisplayStyle = (cssObject: object) => {
+  if (cssObject[':host'].display) {
+    expect(cssObject[':host'].display).not.toMatch(/!important/);
+  } else {
+    // some components don't have a display style
+    expect(cssObject[':host'].display).toBeUndefined();
+  }
+};
+
+// Expect all slotted styles to be !important since they shouldn't be overridable
+const validateSlottedStyles = (cssObject: object, tagName: TagName) => {
+  recursivelyApplyForKey(cssObject, '::slotted', (_, value) => {
+    Object.entries(value).forEach(([cssProp, cssValue]) => {
+      // exceptions for tagName and css property are defined here
+      if (tagName !== 'p-textarea-wrapper' || !['height', 'min-height', 'resize'].includes(cssProp)) {
+        expect(cssValue).toMatch(/ !important$/);
```

Comment:
Maybe remove leading space to discover not using the `addImportantToRule` utility.